lib/auth/oidc.go

```diff
@@ -556,7 +556,7 @@ func claimsFromIDToken(oidcClient *oidc.Client, idToken string) (jose.Claims, er
 func claimsFromUserInfo(oidcClient *oidc.Client, issuerURL string, accessToken string) (jose.Claims, error) {
 	err := isHTTPS(issuerURL)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, trace.NotFound(err.Error())
```

Comment:
Nit: I'd consider adding a comment here explaining that returning NotFound here will make the caller treat this gracefully as a non-existent userinfo endpoint instead of failing auth.